### PR TITLE
Remote reader

### PR DESCRIPTION
**Pull request by emptycrown**
Loader for arbitrary urls (remote pages/files)

### CODE REVIEW

File: loader_hub/file/base.py [hunk 1]
```diff
@@ -20,6 +20,8 @@
     ".epub": "EpubReader",
     ".md": "MarkdownReader",
     ".mbox": "MboxReader",
+    ".eml": "UnstructuredReader",
+    ".html": "UnstructuredReader",
```
Comment (jerryjliu) [inline loader_hub/file/base.py:24]: i'm fine with landing this but this is also outside the scope of this PR right?

File: loader_hub/file/README.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ from langchain.agents import initialize_agent, Tool
 from langchain.llms import OpenAI
 from langchain.chains.conversation.memory import ConversationBufferMemory
 
-SimpleDirectoryReader = download_loader(SimpleDirectoryReader)
+SimpleDirectoryReader = download_loader("SimpleDirectoryReader")
```
Comment (jerryjliu) [inline loader_hub/file/README.md:45]: nice catch

File: loader_hub/remote/base.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            text = "\n\n".join([str(el.decode("utf-8-sig")) for el in result])
+            documents = [Document(text, extra_info=extra_info)]
+        else:
+            suffix = Path(urlparse(url).path).suffix
+            with tempfile.TemporaryDirectory() as temp_dir:
+                filepath = f"{temp_dir}/temp{suffix}"
+                with open(filepath, "wb") as output:
+                    output.write(result.read())
+
+                SimpleDirectoryReader = download_loader("SimpleDirectoryReader")
```
Comment (jerryjliu) [inline loader_hub/remote/base.py:49]: yeah this works

**Review by jerryjliu — APPROVED** [review]

File: loader_hub/file/base.py (hunk 1 above)
Comment (emptycrown) [reply loader_hub/file/base.py:24]: I wanted to make sure that if html files are downloaded, they are parsed correctly since `SimpleDirectoryReader` is used

Comment (jerryjliu) [reply loader_hub/file/base.py:24]: ah right
